tests/libvmifact/factory.go

```diff
@@ -146,3 +148,23 @@ func NewWindows(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 	vmi.Spec.Domain.Firmware = &kvirtv1.Firmware{UUID: WindowsFirmware}
 	return vmi
 }
+
+func NewTinyOSVMWithDataVolume(dv *cdiv1.DataVolume, extraVMOpts ...libvmi.VMOption) *kvirtv1.VirtualMachine {
+	// Supplied with no user data, AlpimeWithTestTooling image takes more than 200s to allow login
+	withNonEmptyUserData := libvmi.WithCloudInitNoCloudEncodedUserData("#!/bin/bash\necho hello\n")
+	opts := []libvmi.VMOption{
+		libvmi.WithDataVolumeTemplate(dv),
+	}
+	opts = append(opts, extraVMOpts...)
+
+	return libvmi.NewVirtualMachine(
+		libvmi.New(
+			libvmi.WithDataVolume("disk0", dv.Name),
+			libvmi.WithResourceMemory(cirrosMemory()),
+			withNonEmptyUserData,
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
```

Comment:
Adding this as input on every call is not that bad. Do you really need it on each call?